make.bat

```diff
@@ -167,22 +275,46 @@ exit /b 1
 :success
 echo  ^> V built successfully!
 echo  ^> To add V to your PATH, run `.\v.exe symlink`.
-del v_old.exe >>%log_file% 2>>&1
-del %log_file%
+if !valid_cc! EQU 0 (
+    echo.
+    echo WARNING:  No C compiler was detected in your PATH. `tcc` was used temporarily
+    echo           to build V, but it may have some bugs and may not work in all cases.
+    echo           A more advanced C compiler like GCC or MSVC is recommended.
+    echo           https://github.com/vlang/v/wiki/Installing-a-C-compiler-on-Windows
+    echo.
+)
+
+del v_old.exe>>!log_file! 2>>&1
+del !log_file!
 
 :version
 echo.
 echo | set /p="V version: "
 .\v.exe version
-if "%cloned_tcc%" NEQ "" (
-	if "%force_tcc%" == "" (
-		echo.
-		echo WARNING:  No C compiler was detected in your PATH. `tcc` was used temporarily
-		echo           to build V, but it may have some bugs and may not work in all cases.
-		echo           A more advanced C compiler like GCC or MSVC is recommended.
-		echo           https://github.com/vlang/v/wiki/Installing-a-C-compiler-on-Windows
-		echo.
-	)
-)
-
 popd
+exit /b 0
+
+:usage
+echo.
+echo Usage:
+echo     %~nx0 [compiler] [options]
+echo.
+echo Compiler :
+echo     -msvc ^| -gcc ^| -[fresh-]tcc ^| -clang    Set C compiler
+echo.
+echo Options:
+echo     -local                        Use the local vc repository without
+echo                                   syncing with remote
+echo     -logfile PATH                 Use the specified PATH as the log
+echo                                   file
+echo     -v ^| --verbose                Output compilation commands to stdout
+echo     -h ^| --help             	  Display this help message and exit
+echo.
+echo Examples:
+echo     %~nx0 -msvc
```

Comment:
what does `%~nx0` do ?

Reply:
Would not it be more readable/maintainable to just put a hardcoded `make` instead?

Reply:
`%~n` expands to a filename only, `%~x` expands to an extension only, so chaining both result in `%~nx0` expansion of `%0`, the full path of the batch script, returns only the filename plus extension, so `make.bat`

Reply:
hmm, ok. I thought usually it is a good practice to keep things modular, I will change it.